apps/etterlatte-behandling/src/main/kotlin/grunnlagsendring/doedshendelse/mellomAttenOgTjueVedReformtidspunkt/OpprettDoedshendelseService.kt

```diff
@@ -0,0 +1,165 @@
+                inTransaction {
+                    lagreDoedshendelserForNyeBeroerte(gyldigeDoedshendelserForAvdoed, beroerteBarn, avdoed, Endringstype.OPPRETTET)
+                }
+            }
+        }
+    }
+
+    private fun lagreDoedshendelserForNyeBeroerte(
+        doedshendelserForAvdoed: List<DoedshendelseInternal>,
+        beroerte: List<PersonFnrMedRelasjon>,
+        avdoed: PersonDTO,
+        endringstype: Endringstype,
+    ) {
+        val eksisterendeBeroerte = doedshendelserForAvdoed.map { it.beroertFnr }
+        val nyeBeroerte =
+            beroerte
+                .map { PersonFnrMedRelasjon(it.fnr, it.relasjon) }
+                .filter { !eksisterendeBeroerte.contains(it.fnr) }
+        nyeBeroerte.forEach { person ->
+            doedshendelseDao.opprettDoedshendelse(
+                DoedshendelseInternal.nyHendelse(
+                    avdoedFnr = avdoed.foedselsnummer.verdi.value,
+                    avdoedDoedsdato = avdoed.doedsdato!!.verdi,
+                    beroertFnr = person.fnr,
+                    relasjon = person.relasjon,
+                    endringstype = endringstype,
+                ),
+            )
+        }
+    }
+
+    private fun lagreDoedshendelser(
+        beroerte: List<PersonFnrMedRelasjon>,
+        avdoed: PersonDTO,
+        endringstype: Endringstype,
+    ) {
+        val avdoedFnr = avdoed.foedselsnummer.verdi.value
+        beroerte.forEach { person ->
+            doedshendelseDao.opprettDoedshendelse(
+                DoedshendelseInternal.nyHendelse(
+                    avdoedFnr = avdoedFnr,
+                    avdoedDoedsdato = avdoed.doedsdato!!.verdi,
+                    beroertFnr = person.fnr,
+                    relasjon = person.relasjon,
+                    endringstype = endringstype,
+                ),
+            )
+        }
+        doedshendelseDao.opprettDoedshendelse(
+            DoedshendelseInternal.nyHendelse(
+                avdoedFnr = avdoedFnr,
+                avdoedDoedsdato = avdoed.doedsdato!!.verdi,
+                beroertFnr = avdoedFnr,
+                relasjon = Relasjon.AVDOED,
+                endringstype = endringstype,
+            ),
+        )
+    }
+
+    private fun finnBeroerteBarn(avdoed: PersonDTO): List<PersonFnrMedRelasjon> =
+        with(avdoed.avdoedesBarn ?: emptyList()) {
+            this
+                .filter { barn -> barn.doedsdato == null }
+                .filter { barn -> barn.mellom18og20PaaDato(REFORMTIDSPUNKT) }
+                .map { PersonFnrMedRelasjon(it.foedselsnummer.value, Relasjon.BARN) }
+        }
+
+    private fun Person.mellom18og20PaaDato(dato: LocalDate): Boolean {
+        // Dersom vi ikke har en fødselsdato antar vi at personen kan ha bursdag på nyttårsaften,
+        // for å sikre at vi får med alle som er under 20 år.
+        val benyttetFoedselsdato = foedselsdato ?: LocalDate.of(foedselsaar, 12, 31)
```

Comment:
Blir denne logikken riktig for gjenoppretting også?  🤔

Reply:
Vil tro dette blir riktig også her. Dersom man har fødselsår satt til 2005 eller 2004 vil personen antatt å være mellom 18 og 20 år på reformtidspunkt.